### PR TITLE
Support keep_timestamp_field in Sinks

### PR DESCRIPTION
**Pull request by yunfengzhou-hub**
## What is the purpose of the change

This pull request adds `keep_timestamp_field` in Sinks' constructors. The default value of this parameter is True, and the corresponding behavior matches the existing implementation. When the parameter is set to False, timestamp field would be removed before persisting to external systems using the Sinks.

## Brief change log

- Add `keep_timestamp_field` in Sinks' constructor
- Modify the signature of methods related to materialization, changing the data type of the sink parameter to Sink class.
- Fix the problem that processing time attribute is left unremoved after Redis lookup join.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: yes

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? python docs

### CODE REVIEW

**Comment by yunfengzhou-hub** [comment]
Hi @lindong28, could you please take a look at this PR?

File: python/feathub/feature_tables/tests/test_redis_source_sink.py [hunk 1]
```diff
@@ -404,6 +404,65 @@ def _test_redis_sink_custom_key(
...
 
 
+def _test_redis_sink_not_keep_timestamp(
+    self: FeathubITTestBase,
+    host: str,
+    port: int,
+    mode: str,
+    redis_client: Union[Redis, RedisCluster],
+):
+    # TODO: Fix the bug that in flink processor when column "val"
```
Comment (lindong28) [inline python/feathub/feature_tables/tests/test_redis_source_sink.py:414]: Can you create a JIRA and provide JIRA in this comment?

**Review by lindong28 — COMMENTED** [review]
Thanks for the PR. LGTM overall. Left one minor comment.

File: python/feathub/feature_tables/tests/test_redis_source_sink.py (hunk 1 above)
Comment (yunfengzhou-hub) [reply python/feathub/feature_tables/tests/test_redis_source_sink.py]: It seems that this problem no longer exist in Feathub after previous code updates and Flink version upgrade. I have removed all such TODOs and add test case that verify output results when the input contains None values.

**Comment by lindong28** [comment]
Thanks for the update. LGTM.
